inc/network/includes-network.php

```diff
@@ -398,8 +402,11 @@ function network_step2( $errors = false ) {
 	$wp_siteurl_subdir = preg_replace( '#^' . preg_quote( $home_path, '#' ) . '#', '', $abspath_fix );
 	$rewrite_base      = ! empty( $wp_siteurl_subdir ) ? ltrim( trailingslashit( $wp_siteurl_subdir ), '/' ) : '';
 
+	$config_filename = 'wp-config.php';
+	$config_filename = apply_filters( 'pantheon_wp_multisite_config_filename', $config_filename );
```

Comment:
This could be simplified to a single line. I also prefer using `.`s to quasi-namespace hook names.

```suggestion
	$config_filename = apply_filters( 'pantheon.multisite.config_filename', 'wp-config.php' );
```

In this case, I would use `pantheon` to identify that it's our custom code, `multisite` to indicate the part of the functionality and `config_filename` to echo back the variable that we're filtering. The `_wp_` is superfluous IMO.